src/script/descriptor.cpp

```diff
@@ -1886,6 +1915,42 @@ std::unique_ptr<DescriptorImpl> ParseScript(uint32_t& key_exp_index, Span<const
         error = "Can only have rawnode() inside tr()";
         return nullptr;
     }
+    if (ctx == ParseScriptContext::P2TR && Func("rawleaf", expr)) {
+        auto arg1 = Expr(expr);
+        std::string leaf_script_str(arg1.begin(), arg1.end());
+        if (!IsHex(leaf_script_str)) {
+            error = "Leaf Script is not hex";
+            return nullptr;
+        }
+        auto leaf_script_bytes = ParseHex(leaf_script_str);
+        CScript leaf_script(leaf_script_bytes.begin(), leaf_script_bytes.end());
+
+        if (!Const(",", expr)) {
+            // Leaf version not specified, return early
+            return std::make_unique<RawLeafDescriptor>(leaf_script, TAPROOT_LEAF_TAPSCRIPT);
+        }
+
+        // Read and process leaf version
+        auto arg2 = Expr(expr);
+        std::string leaf_version_str(arg2.begin(), arg2.end());
+        auto leaf_version = TryParseHex<uint8_t>(leaf_version_str);
+        if (!leaf_version.has_value()) {
+            error = "Leaf Version is not hex";
+            return nullptr;
+        }
+        if (leaf_version->size() > 1) {
+            error = "Leaf Version is too large";
+            return nullptr;
+        }
+        if (leaf_version->size() == 0) {
+            error = "Expected Leaf Version but not provided";
```

Comment:
In 6908349fa559c16164c76fff2f140a39f52279e2 "descriptor: Implement rawleaf(() partial descriptor": Please correct me if I'm wrong, but is this check needed? If the expected leaf version is not provided, won't it be verified before (in `if (!Const(",", expr))`?.

Reply:
Thanks for the review @brunoerg. I pushed https://github.com/bitcoin/bitcoin/pull/30243/commits/abe28609c0c7124c8ecf3f09adb537e3b642d50c which adds a test that triggers this error

Reply:
Cool